pkg/cmd/roachtest/test/test_interface.go

```diff
@@ -26,6 +28,24 @@ const DefaultCockroachPath = "./cockroach"
 // in that environment variable.
 var EnvAssertionsEnabledSeed = "ROACHTEST_ASSERTIONS_ENABLED_SEED"
 
+// EnvWorkloadDurationFlag - environment variable to override
+// default run time duration of workload set in tests
+// usage ROACHTEST_PERF_WORKLOAD_DURATION="5m"
+const EnvWorkloadDurationFlag = "ROACHTEST_PERF_WORKLOAD_DURATION"
+
+// ValidateWorkloadDurationFlag checks if EnvWorkloadDurationFlag
+// input is a valid duration string - matches the pattern
+func ValidateWorkloadDurationFlag(input string) bool {
+	// Define the regex pattern for EnvWorkloadDurationFlag
+	pattern := `^\d+[mhsMHS]$`
+
+	// Compile the regex
+	re := regexp.MustCompile(pattern)
```

Comment:
Nit: this can be lifted into a global to avoid regex compilation on every call. (It also helps the optimizer to inline; `MustCompile` is too large to be inlined, but it may also prevent enclosing funcs from being inlined.)